File: src/test/java/tool/cmnclslib/cls/UnitTest_ClsCmmnArgs.java
package tool.cmnclslib.cls;

import java.io.File;
import java.util.HashMap;
import java.util.Map;
import org.junit.Test;
import tool.cmnclslib.ifc.ICmnLogger;
import tool.cmnclslib.mdl.MdlConst;
import tool.cmnclslib.mdl.MdlFile;
import static org.junit.Assert.*;

/**
 * ClsCmmnArgs の単体テストクラスです。
 */
public class UnitTest_ClsCmmnArgs {

    private final String tempDir = new File(new File(new File(System.getProperty("java.io.tmpdir"), "UnitTest"), "CmnClsLib"), "ClsCmmnArgs").getPath();

    private ClsCmmnArgs createArgs() {
        ICmnLogger logger = new ClsLogger();
        return new ClsCmmnArgs(logger);
    }

    @Test
    public void testInitializeLists() {
        ClsCmmnArgs cmmnArgs = createArgs();
        cmmnArgs.initializeLists();
        assertTrue(cmmnArgs.getKeyNameOfUsernameList().contains("username"));
        assertTrue(cmmnArgs.getKeyNameOfPasswordList().contains("password"));
    }

    @Test
    public void testGetModuleInfo() {
        ClsCmmnArgs cmmnArgs = createArgs();
        boolean result = cmmnArgs.getModuleInfo();
        assertTrue(result);
        assertFalse(cmmnArgs.getExeBaseName().isEmpty());
        assertTrue(cmmnArgs.getPid() > 0);
    }

    @Test
    public void testSplitUserAndDomain() {
        ClsCmmnArgs cmmnArgs = createArgs();
        cmmnArgs.setUsername("TESTDOMAIN\\TestUser");
        cmmnArgs.splitUserAndDomain();

        assertEquals("TESTDOMAIN", cmmnArgs.getDomainName());
        assertEquals("TestUser", cmmnArgs.getUsernameWithoutDomain());
    }

    @Test
    public void testGetArgsForUser() {
        ClsCmmnArgs cmmnArgs = createArgs();
        Map<String, String> map = new HashMap<>();
        map.put("u", "AdminUser");
        map.put("domain", "CORPDOMAIN");
        cmmnArgs.setNamedArgs(map);

        boolean ok = cmmnArgs.getArgsForUser();
        assertTrue(ok);
        assertEquals("AdminUser", cmmnArgs.getUsernameWithoutDomain());
        assertEquals("CORPDOMAIN", cmmnArgs.getDomainName());
    }

    @Test
    public void testReplaceByDictionary() {
        ClsCmmnArgs cmmnArgs = createArgs();
        cmmnArgs.getReplaceDictionary().put("__ENV_ID__", "PRODUCTION");

        String replaced = cmmnArgs.replaceByDictionary("C:\\Data\\__ENV_ID__\\log.txt");
        assertEquals("C:\\Data\\PRODUCTION\\log.txt", replaced);
    }

    @Test
    public void testGetPathParam() {
        MdlFile.deleteRecursively(tempDir);

        ClsCmmnArgs cmmnArgs = createArgs();
        cmmnArgs.getNamedArgs().put("ldir", tempDir);

        String path = cmmnArgs.getPathParam("ldir", MdlFile.PATH_IS_DIRECTORY, true);
        assertEquals(tempDir, path);
        assertTrue(new File(tempDir).exists());

        MdlFile.deleteRecursively(tempDir);
    }

    @Test
    public void testGetCommonArgs() {
        ClsCmmnArgs cmmnArgs = createArgs();
        Map<String, String> map = new HashMap<>();
        map.put("v", "3");
        map.put("force", "");
        map.put("diff", "2");
        cmmnArgs.setNamedArgs(map);

        boolean ok = cmmnArgs.getCommonArgs();
        assertTrue(ok);
        assertEquals(3, cmmnArgs.getVerbose());
        assertTrue(cmmnArgs.isForce());
        assertTrue(cmmnArgs.isDiff());
        assertEquals(2, cmmnArgs.getDiffLevel());
    }

    @Test
    public void testDecryptKeyAndPassword() {
        ClsCmmnArgs cmmnArgs = createArgs();
        cmmnArgs.setDecodeKey(false);
        cmmnArgs.setDecodePasswd(false);

        boolean ok = cmmnArgs.decryptKeyAndPassword();
        assertTrue(ok);
    }

    @Test
    public void testWriteLine() {
        ClsCmmnArgs cmmnArgs = createArgs();
        cmmnArgs.writeLine(MdlConst.LVL_NONE, "Test message");
    }

    @Test
    public void testShowUsage() {
        ClsCmmnArgs cmmnArgs = createArgs();
        cmmnArgs.showUsage();
    }

    @Test
    public void testNewModernMethods() {
        ClsCmmnArgs cmmnArgs = createArgs();
        cmmnArgs.setUserNoDomain("testuser");
        assertEquals("testuser", cmmnArgs.getUserNoDomain());

        cmmnArgs.getUserKeyNames().add("custom_user");
        assertTrue(cmmnArgs.getUserKeyNames().contains("custom_user"));

        cmmnArgs.getPassKeyNames().add("custom_pass");
        assertTrue(cmmnArgs.getPassKeyNames().contains("custom_pass"));

        cmmnArgs.getEncPassKeyNames().add("custom_ep");
        assertTrue(cmmnArgs.getEncPassKeyNames().contains("custom_ep"));

        cmmnArgs.getEncKeyNames().add("custom_key");
        assertTrue(cmmnArgs.getEncKeyNames().contains("custom_key"));

        cmmnArgs.getEncEncKeyNames().add("custom_ek");
        assertTrue(cmmnArgs.getEncEncKeyNames().contains("custom_ek"));

        cmmnArgs.getEncKeySizeNames().add("custom_s");
        assertTrue(cmmnArgs.getEncKeySizeNames().contains("custom_s"));

        cmmnArgs.getReplaceMap().put("__TEST__", "REPLACED");
        assertEquals("REPLACED", cmmnArgs.replaceByMap("__TEST__"));

        cmmnArgs.getShortMap().put("custom", "CUSTOM");
        assertEquals("CUSTOM", cmmnArgs.getShortMap().get("custom"));

        cmmnArgs.getAuthDefMap().put("key", "val");
        assertEquals("val", cmmnArgs.getAuthDefMap().get("key"));

        assertTrue(cmmnArgs.decryptKeyAndPass());
    }
}

File: TestProject1/Class/UnitTest_ClsCmmnArgs.cs
using Xunit;
using Assert = Xunit.Assert;
using CmnClsLib.Class;
using CmnClsLib.Interface;
using CmnClsLib.Module;

namespace TestProject1.Class
{
    public class UnitTest_ClsCmmnArgs
    {
        private readonly string _tempDir = Path.Combine(Path.GetTempPath(), @"UnitTest", "CmnClsLib", "ClsCmmnArgs");

        private ClsCmmnArgs CreateArgs()
        {
            ICmnLogger logger = new ClsLogger();
            return new ClsCmmnArgs(logger);
        }

        [Fact]
        public void Test_InitializeLists()
        {
            var cmmnArgs = CreateArgs();
            cmmnArgs.InitializeLists();
            Assert.Contains("username", cmmnArgs.KeyNameOfUsernameList);
            Assert.Contains("password", cmmnArgs.KeyNameOfPasswordList);
        }

        [Fact]
        public void Test_GetModuleInfo()
        {
            var cmmnArgs = CreateArgs();
            bool result = cmmnArgs.GetModuleInfo();
            Assert.True(result);
            Assert.NotEmpty(cmmnArgs.ExeBaseName);
            Assert.True(cmmnArgs.Pid > 0);
        }

        [Fact]
        public void Test_SplitUserAndDomain()
        {
            var cmmnArgs = CreateArgs();
            cmmnArgs.Username = @"TESTDOMAIN\TestUser";
            cmmnArgs.SplitUserAndDomain();

            Assert.Equal("TESTDOMAIN", cmmnArgs.DomainName);
            Assert.Equal("TestUser", cmmnArgs.UsernameWithoutDomain);
        }

        [Fact]
        public void Test_GetArgsForUser()
        {
            var cmmnArgs = CreateArgs();
            cmmnArgs.NamedArgs = new Dictionary<string, string>
            {
                { "u", "AdminUser" },
                { "domain", "CORPDOMAIN" }
            };

            bool ok = cmmnArgs.GetArgsForUser();
            Assert.True(ok);
            Assert.Equal("AdminUser", cmmnArgs.UsernameWithoutDomain);
            Assert.Equal("CORPDOMAIN", cmmnArgs.DomainName);
        }

        [Fact]
        public void Test_ReplaceByDictionary_And_ObsoleteReplaceByDic()
        {
            var cmmnArgs = CreateArgs();
            cmmnArgs.ReplaceDic["__ENV_ID__"] = "PRODUCTION";

            string replaced1 = cmmnArgs.ReplaceByDictionary(@"C:\Data\__ENV_ID__\log.txt");
            Assert.Equal(@"C:\Data\PRODUCTION\log.txt", replaced1);
        }

        [Fact]
        public void Test_GetPathParam()
        {
            if (Directory.Exists(_tempDir))
            {
                Directory.Delete(_tempDir, true);
            }

            var cmmnArgs = CreateArgs();
            cmmnArgs.NamedArgs["ldir"] = _tempDir;

            string path = cmmnArgs.GetPathParam("ldir", MdlFile.PATH_IS_DIRECTORY, true);
            Assert.Equal(_tempDir, path);
            Assert.True(Directory.Exists(_tempDir));

            if (Directory.Exists(_tempDir))
            {
                Directory.Delete(_tempDir, true);
            }
        }

        [Fact]
        public void Test_GetCommonArgs()
        {
            var cmmnArgs = CreateArgs();
            cmmnArgs.NamedArgs = new Dictionary<string, string>
            {
                { "v", "3" },
                { "force", "" },
                { "diff", "2" }
            };

            bool ok = cmmnArgs.GetCommonArgs();
            Assert.True(ok);
            Assert.Equal(3, cmmnArgs.Verbose);
            Assert.True(cmmnArgs.IsForce);
            Assert.True(cmmnArgs.IsDiff);
            Assert.Equal(2, cmmnArgs.DiffLevel);
        }

        [Fact]
        public void Test_DecryptKeyAndPassword_And_ObsoleteWrapper()
        {
            var cmmnArgs = CreateArgs();
            cmmnArgs.IsDecodeKey = false;
            cmmnArgs.IsDecodePasswd = false;

            bool ok1 = cmmnArgs.DecryptKeyAndPassword();
            Assert.True(ok1);
        }

        [Fact]
        public void Test_WriteLine_And_ObsoleteWriteln()
        {
            var cmmnArgs = CreateArgs();
            cmmnArgs.WriteLine(MdlConst.LVL_NONE, "Test message");
        }

        [Fact]
        public void Test_ShowUsage_And_ObsoleteUsage()
        {
            var cmmnArgs = CreateArgs();
            cmmnArgs.ShowUsage();
        }
    }
}
